packages/yoastseo/src/languageProcessing/researches/findKeyphraseInSEOTitle.js

```diff
@@ -72,6 +73,141 @@ const adjustPosition = function( title, position ) {
 	return position;
 };
 
+/**
+ * Creates a cartesian product of the given arrays.
+ * This function is taken from: https://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript
+ *
+ * @param {array} arrays The arrays to create the cartesian product of.
+ *
+ * @returns {array} The cartesian product of the given arrays.
+ */
+function cartesian( ...arrays ) {
+	return arrays.reduce( ( a, b ) => a.flatMap( d => b.map( e => [ d, e ].flat() ) ) );
+}
```

Comment:
This function is taken from from: https://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript